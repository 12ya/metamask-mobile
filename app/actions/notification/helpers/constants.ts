enum notificationsAction {
  PERFORM_SIGN_IN_REQUEST = 'notifications/perform_sign_in_request',
  PERFORM_SIGN_IN_SUCCESS = 'notifications/perform_sign_in_success',
  PERFORM_SIGN_IN_FAILURE = 'notifications/perform_sign_in_failure',
  PERFORM_SIGN_OUT_REQUEST = 'notifications/perform_sign_out_request',
  PERFORM_SIGN_OUT_SUCCESS = 'notifications/perform_sign_out_success',
  PERFORM_SIGN_OUT_FAILURE = 'notifications/perform_sign_out_failure',
  ENABLE_PROFILE_SYNCING_REQUEST = 'notifications/enable_profile_syncing_request',
  ENABLE_PROFILE_SYNCING_SUCCESS = 'notifications/enable_profile_syncing_success',
  ENABLE_PROFILE_SYNCING_FAILURE = 'notifications/enable_profile_syncing_failure',
  DISABLE_PROFILE_SYNCING_REQUEST = 'notifications/disable_profile_syncing_request',
  DISABLE_PROFILE_SYNCING_SUCCESS = 'notifications/disable_profile_syncing_success',
  DISABLE_PROFILE_SYNCING_FAILURE = 'notifications/disable_profile_syncing_failure',
  ENABLE_PUSH_NOTIFICATIONS_REQUEST = 'notifications/enable_push_notifications_request',
  ENABLE_PUSH_NOTIFICATIONS_SUCCESS = 'notifications/enable_push_notifications_success',
  ENABLE_PUSH_NOTIFICATIONS_FAILURE = 'notifications/enable_push_notifications_failure',
  DISABLE_PUSH_NOTIFICATIONS_REQUEST = 'notifications/disable_push_notifications_request',
  DISABLE_PUSH_NOTIFICATIONS_SUCCESS = 'notifications/disable_push_notifications_success',
  DISABLE_PUSH_NOTIFICATIONS_FAILURE = 'notifications/disable_push_notifications_failure',
  CHECK_ACCOUNTS_PRESENCE_REQUEST = 'notifications/check_accounts_presence_request',
  CHECK_ACCOUNTS_PRESENCE_SUCCESS = 'notifications/check_accounts_presence_success',
  CHECK_ACCOUNTS_PRESENCE_FAILURE = 'notifications/check_accounts_presence_failure',
  DELETE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_REQUEST = 'notifications/delete_on_chain_triggers_by_account_request',
  DELETE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_SUCCESS = 'notifications/delete_on_chain_triggers_by_account_success',
  DELETE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_FAILURE = 'notifications/delete_on_chain_triggers_by_account_failure',
  UPDATE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_REQUEST = 'notifications/update_on_chain_triggers_by_account_request',
  UPDATE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_SUCCESS = 'notifications/update_on_chain_triggers_by_account_success',
  UPDATE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_FAILURE = 'notifications/update_on_chain_triggers_by_account_failure',
  SET_FEATURE_ANNOUNCEMENTS_ENABLED_REQUEST = 'notifications/set_feature_announcements_enabled_request',
  SET_FEATURE_ANNOUNCEMENTS_ENABLED_SUCCESS = 'notifications/set_feature_announcements_enabled_success',
  SET_FEATURE_ANNOUNCEMENTS_ENABLED_FAILURE = 'notifications/set_feature_announcements_enabled_failure',
  SET_SNAP_NOTIFICATIONS_ENABLED_REQUEST = 'notifications/set_snap_notifications_enabled_request',
  SET_SNAP_NOTIFICATIONS_ENABLED_SUCCESS = 'notifications/set_snap_notifications_enabled_success',
  SET_SNAP_NOTIFICATIONS_ENABLED_FAILURE = 'notifications/set_snap_notifications_enabled_failure',
  SET_METAMASK_NOTIFICATIONS_FEATURE_SEEN_REQUEST = 'notifications/set_metamask_notifications_feature_seen_request',
  SET_METAMASK_NOTIFICATIONS_FEATURE_SEEN_SUCCESS = 'notifications/set_metamask_notifications_feature_seen_success',
  SET_METAMASK_NOTIFICATIONS_FEATURE_SEEN_FAILURE = 'notifications/set_metamask_notifications_feature_seen_failure',
  FETCH_AND_UPDATE_METAMASK_NOTIFICATIONS_REQUEST = 'notifications/fetch_and_update_metamask_notifications_request',
  FETCH_AND_UPDATE_METAMASK_NOTIFICATIONS_SUCCESS = 'notifications/fetch_and_update_metamask_notifications_success',
  FETCH_AND_UPDATE_METAMASK_NOTIFICATIONS_FAILURE = 'notifications/fetch_and_update_metamask_notifications_failure',
  MARK_METAMASK_NOTIFICATIONS_AS_READ_REQUEST = 'notifications/mark_metamask_notifications_as_read_request',
  MARK_METAMASK_NOTIFICATIONS_AS_READ_SUCCESS = 'notifications/mark_metamask_notifications_as_read_success',
  MARK_METAMASK_NOTIFICATIONS_AS_READ_FAILURE = 'notifications/mark_metamask_notifications_as_read_failure',
  DELETE_NOTIFICATION_STATUS_REQUEST = 'notifications/delete_notification_status_request',
  DELETE_NOTIFICATION_STATUS_SUCCESS = 'notifications/delete_notification_status_success',
  DELETE_NOTIFICATION_STATUS_FAILURE = 'notifications/delete_notification_status_failure',
  SET_PARTICIPATE_IN_META_METRICS_REQUEST = 'notifications/set_participate_in_meta_metrics_request',
  SET_PARTICIPATE_IN_META_METRICS_SUCCESS = 'notifications/set_participate_in_meta_metrics_success',
  SET_PARTICIPATE_IN_META_METRICS_FAILURE = 'notifications/set_participate_in_meta_metrics_failure',
  UPDATE_TRIGGER_PUSH_NOTIFICATIONS_REQUEST = 'notifications/update_trigger_push_notifications_request',
  UPDATE_TRIGGER_PUSH_NOTIFICATIONS_SUCCESS = 'notifications/update_trigger_push_notifications_success',
  UPDATE_TRIGGER_PUSH_NOTIFICATIONS_FAILURE = 'notifications/update_trigger_push_notifications_failure',
  ENABLE_NOTIFICATIONS_SERVICES_REQUEST = 'notifications/enable_notifications_services_request',
  ENABLE_NOTIFICATIONS_SERVICES_SUCCESS = 'notifications/enable_notifications_services_success',
  ENABLE_NOTIFICATIONS_SERVICES_FAILURE = 'notifications/enable_notifications_services_failure',
  DISABLE_NOTIFICATIONS_SERVICES_REQUEST = 'notifications/disable_notifications_services_request',
  DISABLE_NOTIFICATIONS_SERVICES_SUCCESS = 'notifications/disable_notifications_services_success',
  DISABLE_NOTIFICATIONS_SERVICES_FAILURE = 'notifications/disable_notifications_services_failure',
}

export default notificationsAction;
