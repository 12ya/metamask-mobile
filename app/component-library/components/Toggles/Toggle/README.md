# Toggle

Toggle is a component used to represent a switch between true or false.

## Props

This component extends React Native's [Switch](https://reactnative.dev/docs/switch) component.

### `isSelected`

Determines if toggle is selected.

| <span style="color:gray;font-size:14px">TYPE</span> | <span style="color:gray;font-size:14px">REQUIRED</span> | <span style="color:gray;font-size:14px">DEFAULT</span> |
| :-------------------------------------------------- | :------------------------------------------------------ | :----------------------------------------------------- |
| boolean                                             | No                                                      | false                                                  |

## React Native's Switch Props

### `disabled`

If true the user won't be able to toggle the switch.

| <span style="color:gray;font-size:14px">TYPE</span> | <span style="color:gray;font-size:14px">REQUIRED</span> | <span style="color:gray;font-size:14px">DEFAULT</span> |
| :-------------------------------------------------- | :------------------------------------------------------ | :----------------------------------------------------- |
| boolean                                             | No                                                      | false                                                  |

### `onChange`

Invoked with the the change event as an argument when the value changes.

| <span style="color:gray;font-size:14px">TYPE</span> | <span style="color:gray;font-size:14px">REQUIRED</span> |
| :-------------------------------------------------- | :------------------------------------------------------ |
| function                                            | No                                                     |

### `onValueChange`

Invoked with the new value when the value changes.

| <span style="color:gray;font-size:14px">TYPE</span> | <span style="color:gray;font-size:14px">REQUIRED</span> |
| :-------------------------------------------------- | :------------------------------------------------------ |
| function                                            | No                                                     |

## Usage

```javascript
// Replace import with relative path.
import Toggle from 'app/component-library/components/Toggles/Toggle/Toggle';

<Toggle 
  isSelected 
  disabled/>
```
