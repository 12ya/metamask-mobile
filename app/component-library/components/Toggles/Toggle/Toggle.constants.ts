/* eslint-disable import/prefer-default-export */

// Test IDs
export const TOGGLE_TEST_ID = 'toggle';
