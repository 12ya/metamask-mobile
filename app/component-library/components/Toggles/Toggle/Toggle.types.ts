// Third party dependencies.
import { SwitchProps } from 'react-native';

/**
 * Toggle component props.
 */
export interface ToggleProps extends SwitchProps {
  /**
   * Determines if toggle is selected.
   * @default false
   */
  isSelected?: boolean;
}
