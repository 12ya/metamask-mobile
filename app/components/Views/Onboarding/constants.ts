// eslint-disable-next-line import/prefer-default-export
export const onboardingStrategy = {
  privateKey: 'privateKey',
  seedPhrase: 'seedPhrase',
  qrHardware: 'qrHardware',
};
