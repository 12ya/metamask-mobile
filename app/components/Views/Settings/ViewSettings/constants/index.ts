/* eslint-disable @typescript-eslint/no-var-requires */
/* eslint-disable @typescript-eslint/no-require-imports */
/* eslint-disable import/no-commonjs */
const VIEWS: Record<string, any> = {
  'MetaMask Default':
    require('../../../../../components/Views/Wallet/themes/01').default,
  'MetaMask NFTLover':
    require('../../../../../components/Views/Wallet/themes/02').default,
  'MetaMask web3Banking':
    require('../../../../../components/Views/Wallet/themes/03').default,
};

export default VIEWS;
