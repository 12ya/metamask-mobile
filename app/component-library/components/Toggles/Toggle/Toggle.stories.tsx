/* eslint-disable react-hooks/rules-of-hooks */
/* eslint-disable no-console */

// Third party dependencies.
import React from 'react';
import { boolean } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';

// External dependencies.
import { storybookPropsGroupID } from '../../../constants/storybook.constants';

// Internal dependencies.
import Toggle from './Toggle';
import { ToggleProps } from './Toggle.types';

export const getToggleStoryProps = (): ToggleProps => {
  const disabled = boolean('disabled', false, storybookPropsGroupID);

  return {
    disabled,
  };
};

const ToggleStory = () => <Toggle {...getToggleStoryProps()} />;

storiesOf('Component Library / Toggles', module).add('Toggle', ToggleStory);

export default ToggleStory;
