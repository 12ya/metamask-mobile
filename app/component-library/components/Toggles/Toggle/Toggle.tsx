/* eslint-disable react/prop-types */

// Third party dependencies.
import React, { useState } from 'react';
import { Switch } from 'react-native';

// External dependencies.
import { useStyles } from '../../../hooks';

// Internal dependencies.
import styleSheet from './Toggle.styles';
import { ToggleProps } from './Toggle.types';
import { TOGGLE_TEST_ID } from './Toggle.constants';

const Toggle: React.FC<ToggleProps> = ({
  style,
  isSelected = false,
  onValueChange,
  ...props
}) => {
  const [currentValue, setCurrentValue] = useState(isSelected);
  const { styles, theme } = useStyles(styleSheet, { style });

  const valueChangeHandler = () => {
    setCurrentValue(!currentValue);
    onValueChange?.(currentValue);
  };
  return (
    <Switch
      trackColor={{
        true: theme.colors.primary.default,
        // TODO: Replace Hex value with theme.colors.background.moderate when available
        false: '#848C96',
      }}
      style={styles.base}
      ios_backgroundColor={'#848C96'}
      {...props}
      value={currentValue}
      onValueChange={valueChangeHandler}
      testID={TOGGLE_TEST_ID}
    />
  );
};

export default Toggle;
