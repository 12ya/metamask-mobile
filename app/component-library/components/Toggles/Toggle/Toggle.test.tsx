// Third party dependencies.
import React from 'react';
import { shallow } from 'enzyme';

// Internal dependencies.
import Toggle from './Toggle';
import { TOGGLE_TEST_ID } from './Toggle.constants';

describe('Toggle', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Toggle />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correct toggle when isSelected is false', () => {
    const wrapper = shallow(<Toggle />);
    const toggleComponent = wrapper.findWhere(
      (node) => node.prop('testID') === TOGGLE_TEST_ID,
    );
    expect(toggleComponent.props().value).toBe(false);
  });
  it('should render correct toggle when isSelected is true', () => {
    const wrapper = shallow(<Toggle isSelected />);
    const toggleComponent = wrapper.findWhere(
      (node) => node.prop('testID') === TOGGLE_TEST_ID,
    );
    expect(toggleComponent.props().value).toBe(true);
  });
});
