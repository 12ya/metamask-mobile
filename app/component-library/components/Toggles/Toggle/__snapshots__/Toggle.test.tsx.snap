// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle should render correctly 1`] = `
<ForwardRef(Switch)
  ios_backgroundColor="#848C96"
  onValueChange={[Function]}
  style={Object {}}
  testID="toggle"
  trackColor={
    Object {
      "false": "#848C96",
      "true": "#037DD6",
    }
  }
  value={false}
/>
`;
